packages/client/src/components/Screen.svelte

```diff
@@ -15,24 +16,30 @@
   // Get the screen definition for the current route
   $: screenDefinition = $screenStore.activeScreen?.props
 
-  onMount(() => {
-    // Mark the router as loaded whenever the screen mounts
-    if (!$routeStore.routerLoaded) {
-      routeStore.actions.setRouterLoaded()
-    }
+  $: runOnLoadActions(params)
+
+  // Enrich and execute any on load actions.
+  // We manually construct the full context here as this component is the
+  // one that provides the url context, so it is not available in $context yet
+  const runOnLoadActions = params => {
+    const screenState = get(screenStore)
 
-    // Enrich and execute and on load actions.
-    // We manually construct the full context here as this component is the
-    // one that provides the url context, so it is not available in $context yet
-    if ($screenStore.activeScreen?.onLoad && !$builderStore.inBuilder) {
-      const actions = enrichButtonActions($screenStore.activeScreen.onLoad, {
-        ...$context,
+    if (screenState.activeScreen?.onLoad && !get(builderStore).inBuilder) {
+      const actions = enrichButtonActions(screenState.activeScreen.onLoad, {
+        ...get(context),
         url: params,
       })
-      if (actions != null) {
+      if (actions !== null) {
```

Comment:
Let's keep this as `!=` so we account for both `null` and `undefined` values.

Reply:
Ah apologies, missed this, bad copy and paste.